zerver/lib/actions.py

```diff
@@ -2703,6 +2703,39 @@ def check_schedule_message(
     return do_schedule_messages([send_request])[0]
 
 
+def validate_message_edit_payload(
+    message: Message,
+    stream_id: Optional[int],
+    topic_name: Optional[str],
+    propagate_mode: Optional[str],
+    content: Optional[str],
+) -> None:
+    """
+    Checks that the data sent is well-formed. Does not handle editability, permissions etc.
+    """
+    if topic_name is None and content is None and stream_id is None:
+        raise JsonableError(_("Nothing to change"))
+
+    if not message.is_stream_message():
+        if stream_id is not None:
+            raise JsonableError(_("Private messages cannot be moved to streams."))
+
+    if propagate_mode != "change_one" and topic_name is None and stream_id is None:
+        raise JsonableError(_("Invalid propagate_mode without topic edit"))
+
+    if topic_name is not None:
+        topic_name = topic_name.strip()
```

Comment:
This has a subtle potential bug -- we're no longer doing the stripping in the copy of `topic_name` present in `check_update_message`.  

It turns out it's actually OK for this caller, because `REQ_topic` strips the field before it gets here anyway, but I'm re-adding this in the caller since I'm not confident the new caller in the bot system will naturally do that, and it's a bit subtle.